manual/manual_zx.rst

```diff
@@ -0,0 +1,656 @@
+
+.. jupyter-execute::
+
+    from pytket.passes import ZXGraphlikeOptimisation
+
+    # Use the same CCX example from above
+    ZXGraphlikeOptimisation().apply(c)
+    render_circuit_jupyter(c)
+
+The specific nature of optimising circuits via ZX diagrams gives rise to some general advice regarding how to use :py:class:`ZXGraphlikeOptimisation` in compilation sequences and what to expect from its performance:
+
+.. Extraction techniques are not optimal so starting with a well-structured circuit, abstracting away that structure and starting from scratch is likely to increase gate counts; since graphlike form abstracts away Cliffords to focus on non-Cliffords, most likely to give good results on Clifford-dense circuits
+
+* The routine can broadly be thought of as a resynthesis pass: converting to a graphlike ZX diagram completely abstracts away most of the circuit structure and attempts to extract a new circuit from scratch. Coupling this with the difficulty of optimal extraction means that if the original circuit is already well-structured or close to optimal, it is likely that the process of forgetting that structure and trying to extract something new will increase gate counts. Since the graphlike form abstracts away the structure from Clifford gates to focus on the non-Cliffords, it is most likely going to give its best results on very Clifford-dense circuits. Even in cases where this improves on gate counts, it may be the case that the new circuit structure is harder to efficiently route on a device with restricted qubit connectivity, so it is important to consider the context of a full compilation sequence when analysing the benefits of using this routine.
+
+.. Since ZX does resynthesis and completely abstracts away circuit structure, there is little point in running optimisations before ZX
+
+* Similarly, because the convertion to a graphlike ZX diagram completely abstracts away the Clifford gates, there is often little-to-no benefit in running most simple optimisations before applying :py:class:`ZXGraphlikeOptimisation` since it will largely ignore them and achieve the same graphlike form regardless.
+
+.. Extraction is not optimised so best to run other passes afterwards
+
+* The implementation of the extraction routine in pytket follows the steps from Backens et al. [Back2021]_ very closely without optimising the gate sequences as they are produced. It is recommended to run additional peephole optimisation passes afterwards to account for redundancies introduced by the extraction procedure. For example, we can see in the above example that there are many sequences of successive Hadamard gates that could be removed using a pass like :py:class:`RemoveRedundancies`. :py:class:`FullPeepholeOptimise` is a good catch-all that incorporates many peephole optimisations and could further reduce the extracted circuit. 
+
+Advanced Topics
+---------------
+
+C++ Implementation
+==================
+
+.. Use for speed and more control
```

Comment:
Would it make sense here to reiterate that these structures are not quite performance-oriented, and it may be advisable to compare them with quizx?

Reply:
I could argue that the first note block at the top of the page is bold enough that readers should remember that as they are going through the manual. If they choose to continue reading beyond that point, they have committed to accepting the tradeoffs laid out. But on the other hand, this part is specifically about encouraging users to consider faster, lower-level code and so reiterating faster alternatives could still be good.

I'll try to phrase it in a way that doesn't just make this module sound obsolete